Cargo bench code fails to compile

From some code in this repo, most or all of which was added in commit [`a987a562703918849978323ed734bef064367e68`](https://github.com/hyperium/headers/commit/a987a562703918849978323ed734bef064367e68) in 2019, I am inferring that it is intended that one should be able to run `cargo +nightly bench --features nightly` and have benchmarks run. However, on current master (`682d0a6f80edf3c0f037cc034c6328c74c47258c`) with Rust 1.82 (`rustc 1.82.0 (f6e511eec 2024-10-15)`, `cargo 1.82.0 (8f40fc59f 2024-08-21)`), attempting to run `cargo +nightly bench --features nightly` fails with the following compile errors:

<details><summary>Error</summary>

```
error[E0432]: unresolved import `HeaderMapExt`
   --> src/common/mod.rs:96:17
    |
96  |             use HeaderMapExt;
    |                 ^^^^^^^^^^^^ no external crate `HeaderMapExt`
    |
   ::: src/common/content_type.rs:172:5
    |
172 |     bench_header!(bench_plain, ContentType, "text/plain");
    |     ----------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this trait through its public re-export instead
    |
96  |             use crate::HeaderMapExt;
    |                 ~~~~~~~~~~~~~~~~~~~

error[E0432]: unresolved import `HeaderMapExt`
   --> src/common/mod.rs:96:17
    |
96  |             use HeaderMapExt;
    |                 ^^^^^^^^^^^^ no external crate `HeaderMapExt`
    |
   ::: src/common/content_type.rs:173:5
    |
173 |     bench_header!(bench_json, ContentType, "application/json");
    |     ---------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this trait through its public re-export instead
    |
96  |             use crate::HeaderMapExt;
    |                 ~~~~~~~~~~~~~~~~~~~

error[E0432]: unresolved import `HeaderMapExt`
   --> src/common/mod.rs:96:17
    |
96  |               use HeaderMapExt;
    |                   ^^^^^^^^^^^^ no external crate `HeaderMapExt`
    |
   ::: src/common/content_type.rs:174:5
    |
174 | /     bench_header!(
175 | |         bench_formdata,
176 | |         ContentType,
177 | |         "multipart/form-data; boundary=---------------abcd"
178 | |     );
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this trait through its public re-export instead
    |
96  |             use crate::HeaderMapExt;
    |                 ~~~~~~~~~~~~~~~~~~~

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:102:31
    |
102 |                     <$ty as ::Header>::name(),
    |                               ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:172:5
    |
172 |     bench_header!(bench_plain, ContentType, "text/plain");
    |     ----------------------------------------------------- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:115:31
    |
115 |                     <$ty as ::Header>::name(),
    |                               ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:172:5
    |
172 |     bench_header!(bench_plain, ContentType, "text/plain");
    |     ----------------------------------------------------- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:102:31
    |
102 |                     <$ty as ::Header>::name(),
    |                               ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:173:5
    |
173 |     bench_header!(bench_json, ContentType, "application/json");
    |     ---------------------------------------------------------- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:115:31
    |
115 |                     <$ty as ::Header>::name(),
    |                               ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:173:5
    |
173 |     bench_header!(bench_json, ContentType, "application/json");
    |     ---------------------------------------------------------- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:102:31
    |
102 |                       <$ty as ::Header>::name(),
    |                                 ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:174:5
    |
174 | /     bench_header!(
175 | |         bench_formdata,
176 | |         ContentType,
177 | |         "multipart/form-data; boundary=---------------abcd"
178 | |     );
    | |_____- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find crate `Header` in the list of imported crates
   --> src/common/mod.rs:115:31
    |
115 |                       <$ty as ::Header>::name(),
    |                                 ^^^^^^ not found in the list of imported crates
    |
   ::: src/common/content_type.rs:174:5
    |
174 | /     bench_header!(
175 | |         bench_formdata,
176 | |         ContentType,
177 | |         "multipart/form-data; boundary=---------------abcd"
178 | |     );
    | |_____- in this macro invocation
    |
    = help: consider importing one of these traits:
            crate::Header
            headers_core::Header
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
   --> src/common/mod.rs:107:25
    |
107 |                     map.typed_get::<$ty>().unwrap();
    |                         ^^^^^^^^^
    |
   ::: src/common/content_type.rs:172:5
    |
172 |     bench_header!(bench_plain, ContentType, "text/plain");
    |     ----------------------------------------------------- in this macro invocation
    |
   ::: src/map_ext.rs:11:8
    |
11  |     fn typed_get<H>(&self) -> Option<H>
    |        --------- the method is available for `HeaderMap` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
107 |                     map.typed_try_get::<$ty>().unwrap();
    |                         ~~~~~~~~~~~~~

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
   --> src/common/mod.rs:118:33
    |
118 |                 let typed = map.typed_get::<$ty>().unwrap();
    |                                 ^^^^^^^^^
    |
   ::: src/common/content_type.rs:172:5
    |
172 |     bench_header!(bench_plain, ContentType, "text/plain");
    |     ----------------------------------------------------- in this macro invocation
    |
   ::: src/map_ext.rs:11:8
    |
11  |     fn typed_get<H>(&self) -> Option<H>
    |        --------- the method is available for `HeaderMap` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
118 |                 let typed = map.typed_try_get::<$ty>().unwrap();
    |                                 ~~~~~~~~~~~~~

error[E0599]: no method named `typed_insert` found for struct `HeaderMap` in the current scope
    --> src/common/mod.rs:121:25
     |
121  |                     map.typed_insert(typed.clone());
     |                         ^^^^^^^^^^^^
     |
    ::: src/common/content_type.rs:172:5
     |
172  |     bench_header!(bench_plain, ContentType, "text/plain");
     |     ----------------------------------------------------- in this macro invocation
     |
    ::: src/map_ext.rs:6:8
     |
6    |     fn typed_insert<H>(&mut self, header: H)
     |        ------------ the method is available for `HeaderMap` here
     |
     = help: items from traits can only be used if the trait is in scope
help: there is a method `insert` with a similar name, but with different arguments
    --> /home/phoenix/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-1.1.0/src/header/map.rs:1226:5
     |
1226 | /     pub fn insert<K>(&mut self, key: K, val: T) -> Option<T>
1227 | |     where
1228 | |         K: IntoHeaderName,
     | |__________________________^
     = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_insert` is implemented but not in scope; perhaps you want to import it
     |
95   +             use crate::map_ext::HeaderMapExt;
     |

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
   --> src/common/mod.rs:107:25
    |
107 |                     map.typed_get::<$ty>().unwrap();
    |                         ^^^^^^^^^
    |
   ::: src/common/content_type.rs:173:5
    |
173 |     bench_header!(bench_json, ContentType, "application/json");
    |     ---------------------------------------------------------- in this macro invocation
    |
   ::: src/map_ext.rs:11:8
    |
11  |     fn typed_get<H>(&self) -> Option<H>
    |        --------- the method is available for `HeaderMap` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
107 |                     map.typed_try_get::<$ty>().unwrap();
    |                         ~~~~~~~~~~~~~

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
   --> src/common/mod.rs:118:33
    |
118 |                 let typed = map.typed_get::<$ty>().unwrap();
    |                                 ^^^^^^^^^
    |
   ::: src/common/content_type.rs:173:5
    |
173 |     bench_header!(bench_json, ContentType, "application/json");
    |     ---------------------------------------------------------- in this macro invocation
    |
   ::: src/map_ext.rs:11:8
    |
11  |     fn typed_get<H>(&self) -> Option<H>
    |        --------- the method is available for `HeaderMap` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
118 |                 let typed = map.typed_try_get::<$ty>().unwrap();
    |                                 ~~~~~~~~~~~~~

error[E0599]: no method named `typed_insert` found for struct `HeaderMap` in the current scope
    --> src/common/mod.rs:121:25
     |
121  |                     map.typed_insert(typed.clone());
     |                         ^^^^^^^^^^^^
     |
    ::: src/common/content_type.rs:173:5
     |
173  |     bench_header!(bench_json, ContentType, "application/json");
     |     ---------------------------------------------------------- in this macro invocation
     |
    ::: src/map_ext.rs:6:8
     |
6    |     fn typed_insert<H>(&mut self, header: H)
     |        ------------ the method is available for `HeaderMap` here
     |
     = help: items from traits can only be used if the trait is in scope
help: there is a method `insert` with a similar name, but with different arguments
    --> /home/phoenix/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-1.1.0/src/header/map.rs:1226:5
     |
1226 | /     pub fn insert<K>(&mut self, key: K, val: T) -> Option<T>
1227 | |     where
1228 | |         K: IntoHeaderName,
     | |__________________________^
     = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_insert` is implemented but not in scope; perhaps you want to import it
     |
95   +             use crate::map_ext::HeaderMapExt;
     |

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
    |
   ::: src/map_ext.rs:11:8
    |
11  |       fn typed_get<H>(&self) -> Option<H>
    |          --------- the method is available for `HeaderMap` here
   --> src/common/mod.rs:107:25
    |
107 |                       map.typed_get::<$ty>().unwrap();
    |                           ^^^^^^^^^
    |
   ::: src/common/content_type.rs:174:5
    |
174 | /     bench_header!(
175 | |         bench_formdata,
176 | |         ContentType,
177 | |         "multipart/form-data; boundary=---------------abcd"
178 | |     );
    | |_____- in this macro invocation
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
107 |                     map.typed_try_get::<$ty>().unwrap();
    |                         ~~~~~~~~~~~~~

error[E0599]: no method named `typed_get` found for struct `HeaderMap` in the current scope
    |
   ::: src/map_ext.rs:11:8
    |
11  |       fn typed_get<H>(&self) -> Option<H>
    |          --------- the method is available for `HeaderMap` here
   --> src/common/mod.rs:118:33
    |
118 |                   let typed = map.typed_get::<$ty>().unwrap();
    |                                   ^^^^^^^^^
    |
   ::: src/common/content_type.rs:174:5
    |
174 | /     bench_header!(
175 | |         bench_formdata,
176 | |         ContentType,
177 | |         "multipart/form-data; boundary=---------------abcd"
178 | |     );
    | |_____- in this macro invocation
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_get` is implemented but not in scope; perhaps you want to import it
    |
95  +             use crate::map_ext::HeaderMapExt;
    |
help: there is a method `typed_try_get` with a similar name
    |
118 |                 let typed = map.typed_try_get::<$ty>().unwrap();
    |                                 ~~~~~~~~~~~~~

error[E0599]: no method named `typed_insert` found for struct `HeaderMap` in the current scope
     |
    ::: src/map_ext.rs:6:8
     |
6    |       fn typed_insert<H>(&mut self, header: H)
     |          ------------ the method is available for `HeaderMap` here
    --> src/common/mod.rs:121:25
     |
121  |                       map.typed_insert(typed.clone());
     |                           ^^^^^^^^^^^^
     |
    ::: src/common/content_type.rs:174:5
     |
174  | /     bench_header!(
175  | |         bench_formdata,
176  | |         ContentType,
177  | |         "multipart/form-data; boundary=---------------abcd"
178  | |     );
     | |_____- in this macro invocation
     |
     = help: items from traits can only be used if the trait is in scope
help: there is a method `insert` with a similar name, but with different arguments
    --> /home/phoenix/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-1.1.0/src/header/map.rs:1226:5
     |
1226 | /     pub fn insert<K>(&mut self, key: K, val: T) -> Option<T>
1227 | |     where
1228 | |         K: IntoHeaderName,
     | |__________________________^
     = note: this error originates in the macro `bench_header` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `HeaderMapExt` which provides `typed_insert` is implemented but not in scope; perhaps you want to import it
     |
95   +             use crate::map_ext::HeaderMapExt;
     |

Some errors have detailed explanations: E0405, E0432, E0599.
For more information about an error, try `rustc --explain E0405`.
error: could not compile `headers` (lib test) due to 18 previous errors
```

</details>

Overall, all errors here seem to be downstream of import errors. I suspect that the way item paths work changed in some relevant way at some point in time.

--- a/src/common/mod.rs
+++ b/src/common/mod.rs
@@ -93,13 +93,13 @@ macro_rules! bench_header {
         #[cfg(feature = "nightly")]
         mod $mod {
             use super::$ty;
-            use HeaderMapExt;
+            use crate::HeaderMapExt;
 
             #[bench]
             fn bench_decode(b: &mut ::test::Bencher) {
                 let mut map = ::http::HeaderMap::new();
                 map.append(
-                    <$ty as ::Header>::name(),
+                    <$ty as crate::Header>::name(),
                     $value.parse().expect("HeaderValue::from_str($value)"),
                 );
                 b.bytes = $value.len() as u64;
@@ -112,7 +112,7 @@ macro_rules! bench_header {
             fn bench_encode(b: &mut ::test::Bencher) {
                 let mut map = ::http::HeaderMap::new();
                 map.append(
-                    <$ty as ::Header>::name(),
+                    <$ty as crate::Header>::name(),
                     $value.parse().expect("HeaderValue::from_str($value)"),
                 );
                 let typed = map.typed_get::<$ty>().unwrap();

